dataprofiler/profilers/profile_builder.py

```diff
@@ -689,13 +689,18 @@ def tqdm(l):
                     options=structured_options
                 )
                 new_cols.add(col)
-                        
-        # Format the data
-        notification_str = "Finding the Null values in the columns..."
+                
+        # Generate pool and estimate datasize
         pool = None
+        est_data_size = df[:50000].memory_usage(index=False, deep=True).sum()
```

Comment:
should we sample df first?

Reply:
This does just take the first 50k, but it's fast. Below I generate an estimate which basically identifies the avg bytes per row then multiplies by the number of rows. It's pretty accurate, and the estimated space requirements per-process doubles the actual estimates (meaning the margin for error is 50%).